src/ui/search/search.test.tsx

```diff
@@ -0,0 +1,30 @@
+import { describe, expect, test, vi } from 'vitest'
+import { render } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import userEvent from '@testing-library/user-event'
+import { Search } from './search.tsx'
+
+const user = userEvent.setup()
+
+describe('Search', () => {
+  test('Test that component renders an input with the value equal to initial value passed in props', () => {
+    const onSearch = vi.fn(() => null)
+    const { getByTestId } = render(<Search initialSearch="netflix" onSearch={onSearch} />)
+    expect(getByTestId('search-input')).toHaveValue('netflix')
+  })
+  test('Test that after typing to the input and a "click" event on the Submit button, the "onChange" prop is called with proper value', async () => {
```

Comment:
This test name is really long. And adding implementation details to test names is not the best approach:

```suggestion
  it('submits correct search value', async () => {
```